keps/sig-auth/740-service-account-external-signing/README.md

```diff
@@ -9,105 +9,201 @@
 - [Proposal](#proposal)
   - [Preserve existing behavior](#preserve-existing-behavior)
-  - [Updates to API server token generation](#updates-to-api-server-token-generation)
   - [New API](#new-api)
-  - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  - [Plugins] (#plugins)
+  - [Updates for token generation](#updates-for-token-generation)
+  - [Updates for supported public keys](#updates-for-supported-public-keys)
+  - [Notes/Constraints/Caveats](#notesconstraintscaveats)
   - [Risks and Mitigations](#risks-and-mitigations)
-- [Graduation Criteria](#graduation-criteria)
+- [Design details](design-details)
+  - [ExternalJWTSigner RPC](#externaljwtsigner-rpc)
+  - [Test Plan](#test-plan)
+	  - [Prerequisite testing updates](#prerequisite-testing-updates)
+	  - [Unit tests](#unit-tests)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade/Downgrade Strategy](#upgradedowngrade-strategy)
+  - [Scalability](#scalability)
 - [Implementation History](#implementation-history)
 <!-- /toc -->
 
 ## Summary
 
-The Kubernetes API server has always read service account keys from disk as the process starts, and kept them in memory for the duration of the server's lifetime. As the API server can now verify and issue projected volume tokens, it would be advantageous to support external signing and verifying of token data over an API, as well as reading public keys from an API.
+Service account keys are used by kube-apiserver for JWT signing and authentication. Keys are stored on the disk and are loaded by kube-apiserver during process start time. Key management for service account keys is entirely within the cluster and is less flexible. This KEP proposes updates that will allow Kubernetes distributions to integrate with key management solutions of their choice (eg: HSMs, cloud KMSes).  
 
 ## Motivation
 
-For operators who want to regularly rotate the signing and verifying keys for projected volume tokens, the Kubernetes API server must be restarted in order to use a new key. To facilitate easy key rotation, this KEP includes an proposal for a grpc API to support out of process signing and listing of signing keys.
+Enable integration with alternate key management solutions for service account keys. While at it, make the association between kube-apiserver and service account keys more agile.
+
+Benefits are 2 fold:
+1. Ease of rotation.
+  At present, kube-apiserver loads service account keys at process start time and the key remains the same for the life-time of the process. Thus, rotating keys require kube-apiserver restart. External key management eliminates the need to restart.
+2. Enhanced security.
+  kube-apiserver loads service account keys from a file on the disk. Which means that the service account keys are always present on the disk and can be easily accessed by the cluster administrators. With an external signer, we do not need to store the service account keys on the disk. 
```

Comment:
Maybe rephrase as

"This means anyone with privileged access to the kube-apiserver configuration / filesystem could exfiltrate signing material. An external signer that never returns signing material would mitigate this risk."

Reply:
done